ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/Blockchain.java

```diff
@@ -51,6 +51,13 @@ public interface Blockchain {
    */
   Optional<Hash> getSafeBlock();
 
+  /**
+   * Return the hash of the earliest block in the canonical chain.
+   *
+   * @return The hash of the earliest block in the chain.
+   */
+  Optional<Hash> getEarliest();
```

Comment:
Why not just store the block number instead? It looks like everywhere we use this, we just want the block number and not the block hash so end doing another db call to get the block number from the hash.